### DISCUSSION TITLE
Is there a way to modify webpack chunk name in user's browser

### DISCUSSION THREAD

**Discussion by maazsofttech** [webpack/webpack#16103] [Q&A]
i have a react application in `webpack` setting the chunks are getting generated like  

    50b74e3a.chunk.js

every time its got build the hash changed by `webpack `but that is not useful for cache busting 

in browser , there its cached the old chunk file even after chunk hash changed on server 

so for that i am thinking is there a way to change

    ` 50b74e3a.chunk.js`

 to    
  

   ` 50b74e3a.chunk.js?timestamp=browserUserTime`  so this should come from user's machine

by this way it will be changed by client `Javacsript `at loading time which means it will have to load the fresh copy from server

**Comment by vankop** [comment]
probably you have something like
```
main.js
- 50b74e3a.chunk.js
```

so when your chunk changed, `main.js` hash also changed.. so you have 2 options:
- add hash in naming to `main.js` also ( recommended )
- prevent browser to cache `main.js` ( `Cache-control: no-cache` )

**Reply by maazsofttech** [reply]
@vankop  thanks for replying , issue is sometimes  , even when after a build hash changed browser loads the old hash file from cache ,

for that i was thinking to force browser to take the fresh by appending chunk.js?time=timeAtUser'sMachine

also can you please give a hint on how can i prevent browser to cache a specific file like you suggested

**Reply by vankop** [reply]
> also can you please give a hint on how can i prevent browser to cache a specific file like you suggested

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control

> even when after a build hash changed browser loads the old hash file from cache 

no, it is possible only:
- in html code you have old `<script>` tag
- your `main.js` script is old too

**Reply by maazsofttech** [reply]
@vankop  in my index.html i don't have have any script tag regarding these js , 
only when its build the these  chunks are inserted over there by webpack,  
so they are not old , but i think browser is taking the old version from proxy cache

**Reply by vankop** [reply]
> only when its build the these chunks are inserted over there by webpack,
so they are not old 

I dont understand what you mean.. please create small reproducible repo
